Modern/wrapperPlugins/src/main/java/org/bonej/wrapperPlugins/wrapperUtils/UsageReporter.java

```diff
@@ -0,0 +1,294 @@
+	
+/**
+	 * BoneJ version FIXME: it is fragile to have the version hard-coded here.
+	 * Create a BoneJApp instead.
+	 */
+	private static final String BONEJ_VERSION = "2.0.0-experimental";
+	private static final String ga = "http://www.google-analytics.com/__utm.gif?";
+	private static final String utmwv = "utmwv=5.2.5&";
+	private static final String utmhn = "utmhn=bonej.org&";
+	private static final String utmcs = "utmcs=" + Charset.defaultCharset() + "&";
+	private static final String utmac = "utmac=UA-366405-8&";
+	private static final String utmdt = "utmdt=bonej.org%20Usage%20Statistics&";
+	private static final String utmt = "utmt=event&";
+	private static final String utmul = "utmul=" + getLocaleString() + "&";
+	private static final String utmje = "utmje=0&";
+	private static final String utmr = "utmr=-&";
+	private static final String utmp = "utmp=%2Fstats&";
+	private static final Random random = new Random();
+	private static String utmcnr = "";
+	private static String utme;
+	private static String utmn;
+	private static String utmsr;
+	private static String utmvp;
+	private static String utmsc;
+	/** iterated on each new event */
+	private static int session = 0;
+	private static String utms = "utms=" + session + "&";
+	private static String utmcc;
+	private static long thisTime = 0;
+	private static long lastTime = 0;
+	/** iterated each time a new BoneJ session starts*/
+	private static int bonejSession;
+	private static boolean isFirstRun = true;
+
+	private static String utmhid;
+
+	private static UsageReporterOptions uro;
+		
+	private UsageReporter() {}
+
+	/**
+	 * Send the report to Google Analytics in the form of an HTTP request for a
+	 * 1-pixel GIF with lots of parameters set
+	 */
+	public void send() {
+		System.out.println("Sending report.\n");
+		//check if user has opted in and die if user has opted out
+		uro = new UsageReporterOptions();
+		if (!uro.isAllowed()) {
+			System.out.println("Usage reporting forbidden by user\n");
+			return;
+		}
+		try {
+			System.out.println("Usage reporting approved by user, preparing URL");
+			final URL url = new URL(ga + utmwv + utms + utmn + utmhn + utmt + utme +
+				utmcs + utmsr + utmvp + utmsc + utmul + utmje + utmcnr + utmdt +
+				utmhid + utmr + utmp + utmac + utmcc);
+			final URLConnection uc = url.openConnection();
+			uc.setRequestProperty("User-Agent", userAgentString());
+			
+//			if (logger.getLevel() < LogLevel.INFO) {
+////				return;
+//			}
+			
+//			logService.info(url.toString());
+			System.out.println(url.toString()+"\n");
+//			logService.info(uc.getRequestProperty("User-Agent"));
+			System.out.println(uc.getRequestProperty("User-Agent")+"\n");
+			try (final BufferedReader reader = new BufferedReader(
+				new InputStreamReader(uc.getInputStream())))
+			{
+//				logService.info(reader.lines());
+//				System.out.println(reader.lines());
+				reader.lines().forEachOrdered(item -> System.out.println(item));
```

Comment:
Why are we printing to System.out..?

Reply:
Because I needed debug info and didn't manage to get the Logger to work. It's not a long-term solution.